src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -0,0 +1,451 @@
+        case localBackendInfo: LocalCallBackendInfo =>
+          dataAccess.localJobs.filter(
+            _.executionId === executionResult.executionId
+          ).map(cols => (
+            cols.pid,
+            cols.rc
+            )).update(
+              localBackendInfo.processId,
+              localBackendInfo.resultCode
+            )
+
+        case jesBackendInfo: JesCallBackendInfo =>
+          dataAccess.jesJobs.filter(
+            _.executionId === executionResult.executionId
+          ).map(cols => (
+            cols.jesId,
+            cols.jesStatus
+            )).update(
+              jesBackendInfo.jesId,
+              jesBackendInfo.jesStatus
+            )
+
+        case null =>
+          throw new NotImplementedError("Backend info is null")
+      }
+
+      _ = require(backendUpdate == 1, s"Unexpected backend update count $backendUpdate")
+
+    } yield ()
+
+    runTransaction(action)
+  }
+
+  /** Get all inputs for the scope of this call. */
+  override def getInputs(workflowId: WorkflowId, call: Call): Future[Traversable[SymbolStoreEntry]] = {
+
+    val action = for {
+
+      workflowExecutionResult <- dataAccess.workflowExecutions.filter(
+        _.workflowExecutionUuid === workflowId.toString).result.head
+
+      symbolResults <- dataAccess.symbols.filter(symbol =>
+        symbol.workflowExecutionId === workflowExecutionResult.workflowExecutionId &&
+          symbol.io === IoInput &&
+          symbol.scope === call.fullyQualifiedName).result
+
+      symbolStoreEntries = symbolResults map { symbolResult =>
+        val wdlType = WdlType.fromWdlString(symbolResult.wdlType)
+        val symbolStoreKey = new SymbolStoreKey(
+          symbolResult.scope,
+          symbolResult.name,
+          symbolResult.iteration,
+          symbolResult.io == IoInput)
+        new SymbolStoreEntry(
+          symbolStoreKey,
+          wdlType,
+          Option(symbolResult.wdlValue).map(wdlType.coerceRawValue(_).get))
+
+      }
+    } yield symbolStoreEntries
+
+    runTransaction(action)
+  }
+
+  /** Get all outputs for the scope of this call. */
+  override def getOutputs(workflowId: WorkflowId, call: Call): Future[Traversable[SymbolStoreEntry]] = {
+
+    val action = for {
+
+      workflowExecutionResult <- dataAccess.workflowExecutions.filter(
+        _.workflowExecutionUuid === workflowId.toString).result.head
+
+      symbolResults <- dataAccess.symbols.filter(symbol =>
+        symbol.workflowExecutionId === workflowExecutionResult.workflowExecutionId &&
+          symbol.io === IoOutput &&
+          symbol.scope === call.fullyQualifiedName).result
+
+      symbolStoreEntries = symbolResults map { symbolResult =>
+        val wdlType = WdlType.fromWdlString(symbolResult.wdlType)
+        val symbolStoreKey = new SymbolStoreKey(
+          symbolResult.scope,
+          symbolResult.name,
+          symbolResult.iteration,
+          symbolResult.io == IoInput)
```

Comment:
Made this more clear with named paramaters: `input = symbolResult.io == IoInput`